### PR TITLE
Fix/175 176 accumulated tiv

### LINKED ISSUE #175: fmcalc: accumulated tiv duplicated for multi-peril analyses
accumulated_tiv is a variable in fmcalc to sum the TIV in the coverages file according to the agg_id hierarchy in order to apply % tiv financial terms.  When there is a many to 1 relationship of items to coverages (ie multi-peril models), the tiv is being multiplied up by the number of items per coverage. This means the the calculation of % TIV terms is incorrect.

Mostly this affects step policies which have % TIV trigger and or % TIV payout terms. If the TIV is 2-4 times bigger, it means that fewer losses attach to the % TIV trigger points than they should. Also when steps do attach, % TIV payouts would be 2-4 times bigger than they should be.

The standard policy % TIV calcrules are not used in oasislmf. This is because the terms are calculated in advance using the full policy TIV  rather than at runtime, where some TIV may be excluded due to not being in the scope of a model (such as business interruption coverages), or because the exposure is not at risk in the model run (the mean damage is zero).

Two ways to deal with this are;
1) fix the fmcalc  code to properly calculate  accumulated_tiv variable for many-to-1 items to coverages
2) convert %TIV terms to amounts upfront, and change the calcrules to use amounts rather than fractions for step policies as has been done for the standard policy types.

### PR DESCRIPTION
**Pull request by johcarter**
fixes #175 #176

### CODE REVIEW

File: src/fmcalc/fmcalc.cpp [hunk 1]
```diff
@@ -804,7 +804,9 @@ void fmcalc::dofm(int event_id, const std::vector<int> &items, std::vector<vecto
 				agg_vec[vec_idx].agg_id = avx[1][vid].agg_id;
 				agg_vec[vec_idx].item_idx = &avx[1][vid].item_idx;
 				if (isGULStreamType_ == true || stepped_ == true) {
-					agg_vec[vec_idx].accumulated_tiv = item_to_tiv_[avx[1][vid].item_idx[0]+1];
+//					agg_vec[vec_idx].accumulated_tiv = item_to_tiv_[avx[1][vid].item_idx[0] + 1];
+//					agg_vec[vec_idx].accumulated_tiv = item_to_tiv_[avx[1][vid].agg_id];
```
Comment (sstruzik) [inline src/fmcalc/fmcalc.cpp:808]: I would remove all the commented lines 378, 807 and 808

**Review by sstruzik — APPROVED** [review]
Looks good as a quick fix.
we will need to do two things to follow up.
1. with the solution implemented (and it was like that before) we only accumulate tiv of item that are present in the event (so %tiv are not event independent).
2. we need to handle more cleanly the old/new gul stream. At the moment we had to sacrifice the potential error raised in init_itemtotiv (like "FATAL: Item ids are not contiguous or do not start from one") to make it work when it should work in when using new gul stream.

File: src/fmcalc/fmcalc.cpp [hunk 2]
```diff
@@ -1066,11 +1070,15 @@ void fmcalc::init_itemtotiv()
...
+		if (!used_tiv[itm.coverage_id]) {
+			item_to_tiv_[itm.id] = coverages[itm.coverage_id];
+			used_tiv[itm.coverage_id] = true;
+		}
+		else {
+			item_to_tiv_[itm.id] = 0;
+		}
 		i = fread(&itm, sizeof(itm), 1, fin);
 	}
-	fclose(fin);
```
Comment (sstruzik) [inline src/fmcalc/fmcalc.cpp:1072]: I think we still need to close the file

Comment (johcarter) [reply src/fmcalc/fmcalc.cpp:1072]: i haven't changed this bit. what is your suggestion?

**Comment by johcarter** [comment]
fixed #176 except for known issue

**Review by hchagani-oasislmf — APPROVED** [review]
Looks good!

**Pull request merged**